src/modules/scene/name_scene.py

```diff
@@ -26,7 +31,8 @@ def start_scene(self):
 
                 if event.type == pygame.KEYDOWN:
                     if event.key == pygame.K_RETURN:
-                        # TODO(nickbar01234) - Send to server
+                        self.get_network().send_name(name)
+                        self.__network_barrier.release()
```

Comment:
critical change - this fixes a bug where sometimes the name acknowledge/response will arive out of order and cause InvalidMessage exception thrown

Reply:
Good catch!